src/nvim/spell.c

```diff
@@ -3648,7 +3645,8 @@ static void suggest_trie_walk(suginfo_T *su, langp_T *lp, char_u *fword, bool so
   bool goodword_ends;
 #ifdef DEBUG_TRIEWALK
   // Stores the name of the change made at each level.
-  char_u changename[MAXWLEN][80];
+  static const size_t changenameElemSize = 80;
+  char_u changename[MAXWLEN][changenameElemSize];
```

Comment:
The result looks much like VLA. I am not sure what exactly compiler would do, but you must not define dimensions using anything of which compiler not thinks like it is “an integer constant expression”. According to the standard this construct defenitely *is* a VLA:

> An integer constant expression99) shall have integer type and shall only have operands that are integer constants, enumeration constants, character constants, sizeof expressions whose results are integer constants, and floating constants that are the immediate operands of casts. Cast operators in an integer constant expression shall only convert arithmetic types to integer types, except as part of an operand to the sizeof operator.

though compiler may think otherwise.

Also note that `changenameElemSize` is *not* a valid name for variable according to the style guide. For constants (though I never saw it used for constants scoped to function) it is `kChangenameElemSize`, for variables it is `changename_elem_size`.

And you ultimately not need any variable here at all. I have checked how `void main(){char a[10][20]; printf("%zu", sizeof(a[0]));}` program works and it returns 20 on gcc, clang and tcc, so you should use `sizeof` in place of variable. Also note that comment that is now before `changenameElemSize` definition actually applies to `changename` and *not* `changenameElemSize` which is incorrect.

Reply:
I totally missed that this is in a function, so of course the `static` is not needed. Good catch.